### PR TITLE
NEO-865 // add `creatable` prop to Select

### PR DESCRIPTION
**Pull request by joe-s-avaya**
For `searchable` select, also allow users to create their own items.

Will add tests in next PR as this one is already large. 

Example Functionality:

### CODE REVIEW

File: src/components/Select/utils/useDownshift.ts [hunk 1]
```diff
@@ -40,14 +46,30 @@ const DownshiftWithComboboxProps = (
...
         setFilteredOptions(options);
       }
+
+      setInputText(inputValue || "");
     },
     onSelectedItemChange: ({ selectedItem: clickedItem }) => {
       if (!clickedItem) {
         setSelectedItems([]);
+      } else if (creatable && clickedItem.value === createOptionValue) {
+        const createdItem: SelectOptionProps = {
```
Comment (yangAtSpoken) [inline src/components/Select/utils/useDownshift.ts:68]: Why do you have to create a new object? Can you use clickedItem?

**Comment by joe-s-avaya** [comment]
> Looks good. The TextInput files look familiar, are they from another PR?

@yangAtSpoken, correct, you'll notice that those files are no longer there now that I've merged some other PRs.

File: src/components/Select/utils/useDownshift.ts [hunk 2]
```diff
@@ -11,19 +13,23 @@ const DownshiftWithComboboxProps = (
...
+  createMessage: string
 ) => {
+  // HACK: `onSelectedItemChange`'s `inputValue` is always `[object Object]`, no idea why
+  const [inputText, setInputText] = useState("");
+
   return useCombobox({
     items: filteredOptions,
     id: selectId,
-    stateReducer: (state, actionAndChanges) => {
+    stateReducer: (_, actionAndChanges) => {
```
Comment (enrique-prado) [inline src/components/Select/utils/useDownshift.ts:26]: What does this underscore do?

**Review by enrique-prado — APPROVED** [review]
Left a question.

File: src/components/Select/utils/useDownshift.ts (hunk 2 above)
Comment (joe-s-avaya) [reply src/components/Select/utils/useDownshift.ts:26]: tells the linter to ignore it (as it's unused)

**Pull request merged**